prune_shard_list.c

```diff
@@ -39,10 +39,20 @@
 #include "utils/errcodes.h"
 #include "utils/lsyscache.h"
 #include "utils/typcache.h"
+#include "utils/memutils.h"
+
+
+/*
+ * OperatorTypeCache is used for caching base operator types for given typeId,
+ * accessMethodId and strategyNumber. It is initialized to empty list as
+ * there are no items in the cache.
+ */
+static List *OperatorTypeCache = NIL;
 
 
 /* local function forward declarations */
 static Oid GetOperatorByType(Oid typeId, Oid accessMethodId, int16 strategyNumber);
+static Oid LookupOperatorByType(Oid typeId, Oid accessMethodId, int16 strategyNumber);
```

Comment:
Move this up one line. The declaration and definition orders should match, but these declarations are in another order than the definitions.